common/src/main/java/org/opensearch/sdk/SdkClientUtils.java

```diff
@@ -51,4 +54,21 @@ public static Throwable getRethrownExecutionExceptionRootCause(Throwable throwab
         }
         return throwable;
     }
+
+    /**
+     * If an internal variable is an enum represented by all upper case, the Remote client may have it mapped in lower case. This method lowercases these enum values 
+     * @param field The JSON field to lowercase the value
+     * @param json The full JSON to process
+     * @return The JSON with the value lowercased
+     */
+    public static String lowerCaseEnumValues(String field, String json) {
```

Comment:
Can't we use GSON to manipulate the JSON Structure? like:

```
public static String lowerCaseEnumValues(String field, String json) {
    // Parse the JSON string into a JsonObject
    JsonElement jsonElement = JsonParser.parseString(json);
    
    if (jsonElement.isJsonObject()) {
        JsonObject jsonObject = jsonElement.getAsJsonObject();
        JsonElement fieldElement = jsonObject.get(field);

        // Check if the field exists and is a string
        if (fieldElement != null && fieldElement.isJsonPrimitive() && fieldElement.getAsJsonPrimitive().isString()) {
            String value = fieldElement.getAsString();
            
            // If the value is an all-uppercase string, convert it to lowercase
            if (value.matches("[A-Z_]+")) {
                jsonObject.addProperty(field, value.toLowerCase(Locale.ROOT));
            }
        }
    }
    
    // Return the modified JSON as a string
    return new Gson().toJson(jsonElement);
}
```

Reply:
@dhrubo-os we could, but that brings the overhead of entirely parsing the String into a whole tree of Java objects, then re-creating the String, just to do a string replacement on a single value.  

I'm hoping this is temporary code anyway, as it's possible the java client will update to handle this.